Enable dunovo (calling rare variants) tutorial.

NickSto opened the pull request:
This tutorial was written and merged a while ago, but apparently it never got enabled.

@nekrut asked if it could be pushed to the GTN website.

wm75 commented:
Before we do this, could you address a couple of issues in it?

* Is 3 hours total really an adequate time estimate for the tutorial?
  In it, you're mentioning an analysis bottleneck that on its own would take 2 hours. There's also a
  bwa-mem step and quite some others, which leaves me with the impression that 4 hours may be
  more realistic. Of course, you're the expert for this training so feel free to ignore me on this one.
* Your description of getting the data onto another Galaxy instance does not work anymore like this.
  With a recent update to the GUI, you now have to go through the Top bar User menu to import
  histories. Please adjust accordingly.
* Would you mind adding the steps necessary to get the input data directly from the SRA as third
  alternative to the *Importing the raw data* section? It's ok to have the link to the video, but users
  shouldn't be *forced* to watch it.

In addition, could you please:
* add a hands-on box for the FastQC step instead of just referring to the history
* inside hands-on boxes, remove references to specific datasets from the usegalaxy.org history, *e.g.* `18: Naive Variant Caller (NVC) on data 17`?
  Instead use terms like `the output of Naive Variant Caller`.
* mention in the tutorial which steps need to be reconfigured for an SSCS-based analysis instead of *only* providing links to the preconfigured workflows
  Ideally you would teach users how to extract a workflow from a history (or link to a tutorial that shows how), then guide them through the necessary changes step-by-step. You would provide the ready-to-use workflows then only for people in a hurry.

My apologies if that sounds like a lot of requests. I really like the way you're explaining things in this tutorial, but we try to have all contributors to the GTN material adhere to some minimal standards in terms of layout and structure.
Thanks for the effort you are putting into this. It's very much appreciated!

NickSto commented:
@wm75 Thanks for looking through this.
I think the time estimate is actually pretty accurate. I checked how long the BWA step took, and it was 17 seconds! The size of the data is drastically reduced after the consensus sequences are created.
I'll look into the things you mentioned and try to get it in shape.

NickSto commented:
Okay, so I've updated the instructions for importing the data. I've simplified it down to just one method: importing the shared history. The direct SRA import is too problematic and it turns out links to exported history files aren't persistent anyway.

I also made the requested changes to the FastQC step, the dataset references in hands-on boxes, and the description of the SSCS analysis.

Let me know if the changes seem adequate.

By the way, one of the images already in the tutorial is committed as a .png, but it's just a PNG export of an SVG I made in Inkscape. Should I commit the .svg as well, instead of the .png, or both? One possible wrinkle is that the SVG is Inkscape's variant on the format, and I don't think you can reproduce the .png without Inkscape.

shiltemann commented:
Hi @NickSto thanks a lot for working on this! We want to make sure that all our tutorials can be run on any Galaxy instance, so please use [Zenodo](https://zenodo.org/) to store all the input data instead of a history. Then the upload box can be rewritten to use this (see [this tutorial](https://training.galaxyproject.org/training-material/topics/introduction/tutorials/galaxy-intro-101-everyone/tutorial.html#upload-iris-dataset) as an example)

Added value here is that Planemo can be used to automatically extract a `data-library.yaml` file from this zenodo link, that Galaxy admins can use to make the data available as a shared data library on their server (tutorial on how to create this file can be found [here](https://training.galaxyproject.org/training-material/topics/contributing/tutorials/create-new-tutorial-technical/tutorial.html#creating-the-data-libraryyaml-recommended)).

Thanks again, and let me know if you would like some help with any of it!

NickSto commented:
@shiltemann Thanks for taking a look. I've update the text with the Zenodo method.

One note about Zenodo: I uploaded the datasets, but I wasn't sure what to select as a license. It's an SRA upload that I didn't produce. I can't find any IP policy for SRA datasets, and as I'm not the author, I don't have any authority to set a license. To be safe, I just chose "Other (Not Open)".

erasche commented on topics/variant-analysis/tutorials/dunovo/tutorial.md:
```diff
@@ -104,23 +103,41 @@ The starting point of the analysis is sequencing reads (in [FASTQ](https://en.wi
 ## Getting data in and assessing quality
 
-We uploaded the [Schmitt *et al.* 2015](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4414912/) data directly from SRA as shown in [this screencast](https://vimeo.com/121187220).
+You can obtain the data from [Schmitt *et al.* 2015](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4414912/) from Zenodo, or just import the [Galaxy history](https://usegalaxy.org/u/nstoler/h/srr1799908---schmitt-2015) that accompanies this tutorial.
```
```suggestion
You can obtain the data from [Schmitt *et al.* 2015](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4414912/) from [Zenodo](https://doi.org/10.5281/zenodo.3554549), or just import the data from the Galaxy Data library under the "GTN Materials" folder.
```

not sure about the wording here but shouldn't link to a history only available on one server.

erasche commented on topics/variant-analysis/tutorials/dunovo/tutorial.md:
```diff
@@ -3,7 +3,6 @@ layout: tutorial_hands_on
 
 title: "Calling very rare variants"
 zenodo_link: ""
```
```suggestion
zenodo_link: "https://doi.org/10.5281/zenodo.3554549"
```

erasche commented on topics/variant-analysis/tutorials/dunovo/tutorial.md:
```diff
@@ -104,23 +103,41 @@ The starting point of the analysis is sequencing reads (in [FASTQ](https://en.wi
+> ### {% icon hands_on %} Hands-on: Importing the input FASTQ files
 >
-> You can obtain the reads from this dataset by copying [this history](https://usegalaxy.org/u/nstoler/h/srr1799908---schmitt-2015).
+> Option 1: Import the files from Zenodo (this will work with any Galaxy instance):
```
Could you make this hands on box look more like this one? https://github.com/galaxyproject/training-material/blob/master/topics/sequence-analysis/tutorials/quality-control/tutorial.md#inspect-a-raw-sequence-file

Most of the upload steps across the training materials are quite uniform and it might help users to see similar things here.

erasche commented on topics/variant-analysis/tutorials/dunovo/tutorial.md:
```diff
@@ -104,23 +103,41 @@ The starting point of the analysis is sequencing reads (in [FASTQ](https://en.wi
+>    `https://zenodo.org/record/3554549/files/SRR1799908_reverse.fastq?download=1`
+> 6. Then click the *Start* button, then *Close*.
+>
+> Option 2: Obtain the files by importing [this Galaxy history](https://usegalaxy.org/u/nstoler/h/srr1799908---schmitt-2015) (this only works on [usegalaxy.org](https://usegalaxy.org)):
```
Would you remind removing this? It unfortunately works only on one of the servers, and we try really hard to make all of the tutorials work on every server.

You've mentioned `usegalaxy.org` in a couple other places too, is it possible you could search for and replace those references with more generic instructions that will work for everyone? E.g. If you copy the workflows you've created into the training material then we can additionally test them with [automated workflow testing](https://github.com/usegalaxy-eu/workflow-testing/) that can let trainers know they work on specific servers.

erasche commented on topics/variant-analysis/tutorials/dunovo/tutorial.md:
```diff
@@ -332,13 +331,11 @@ Now we'll want to parse the VCF produced by the NVC, determine what the major an
 >
 > Run {% icon tool %} **Variant Annotator** with the following parameters:
->  - {% icon param-file %} *Input variants from Naive Variants Detector*: `18: Naive Variant Caller (NVC) on data 17`
+>  - {% icon param-file %} *Input variants from Naive Variants Detector*: The output of {% icon tool %} **Naive Variant Caller (NVC)**
```
:100: great improvement :)

shiltemann commented:
@NickSto it should be fine putting SRA data on Zenodo ("They are designed to provide and encourage access within the scientific community to sources of current and comprehensive information. Therefore, NCBI itself places no restrictions on the use or distribution of the data contained therein" see [here](https://www.ncbi.nlm.nih.gov/home/about/policies/)).

The tutorial states that one of the steps takes 2 hours, could that be reduced by e.g. down-sampling the input files? (This is what we do for most tutorials, usually is enough to illustrate the analysis and still give some nice results). If not, could you please also add the resulting files from this step to Zenodo (same repo as input data), so that users and/or instructors can continue the analysis without waiting 2 hours if they wish?

shiltemann commented:
@NickSto @nekrut I've applied some of the requested changes myself and made some other enhancements, could you please check this tutorial since the original PR is quite old, anything else that needs to be updated?

NickSto commented:
@shiltemann Sorry, I took a shot at reloading all this into my brain but it turned out to be more than I could find time for right now. Thanks for coming back to this!
